frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,151 @@
+import { createEffect, createSignal } from "solid-js";
+import { createStore } from "solid-js/store";
+import { styled } from "solid-styled-components";
+import Button from "../components/form/button";
+import InputEmail from "../components/form/email-input";
+import InputHash from "../components/form/hash-input";
+import Slider from "../components/slider";
+import InputFile from "../components/form/file-input";
+import { getFilesUrls } from "../helpers/getFilesUrl";
+import { validateEmail } from "../helpers/validation";
+
+const Train = () => {
+  // store of data from inputs of form
+  const [formData, setFormData] = createStore({
+    email: "",
+    files: [],
+  });
+
+  // store of errors inputs of form
+  const [formErrors, setFormErrors] = createStore({
+    email: "",
+    files: "",
+  });
+
+  const [hash, setHash] = createSignal("");
```

Comment:
Why don't you use [`createResource`](https://www.solidjs.com/docs/latest/api#createeffect) to represent the result of the async call?

Reply:
in refactoring result [createResource](https://www.solidjs.com/docs/latest/api#createeffect) not need

Reply:
Please, do not resolve the conversation when we discuss something or you answer/reject my suggestion